spec/unit/http/client_spec.rb

```diff
@@ -45,6 +48,38 @@
         client.connect(uri)
       }.to raise_error(Puppet::HTTP::ConnectionError, %r{^Request to https://www.example.com timed out connect operation after .* seconds})
     end
+
+    it 'connects using the default ssl context' do
+      expect(pool).to receive(:with_connection) do |_, verifier|
+        expect(verifier.ssl_context).to equal(puppet_context)
+      end
+
+      client.connect(uri)
+    end
+
+    it 'connects using a specified ssl context' do
+      other_context = Puppet::SSL::SSLContext.new
+
+      expect(pool).to receive(:with_connection) do |_, verifier|
+        expect(verifier.ssl_context).to equal(other_context)
+      end
+
+      client.connect(uri, ssl_context: other_context)
+    end
+
+    it 'connects using the system store' do
+      expect(pool).to receive(:with_connection) do |_, verifier|
+        expect(verifier.ssl_context).to equal(system_context)
```

Comment:
I guess it didn't like that?

```
  1) Puppet::HTTP::Client when connecting connects using the system store
     Failure/Error: expect(verifier.ssl_context).to equal(system_context)
       expected #<Puppet::SSL::SSLContext:47358080> => #<struct Puppet::SSL::SSLContext store={}, cacerts=[], crls=[], private_key=nil, client_cert=nil, client_chain=[], revocation=true, verify_peer=true>
            got #<Puppet::SSL::SSLContext:47358160> => #<struct Puppet::SSL::SSLContext store={}, cacerts=[], crls=[], private_key=nil, client_cert=nil, client_chain=[], revocation=true, verify_peer=true>
       Compared using equal?, which compares object identity,
       but expected and actual are not the same object. Use
       `expect(actual).to eq(expected)` if you don't care about
       object identity in this example.
     # ./spec/unit/http/client_spec.rb:72:in `block (4 levels) in <top (required)>'
     # ./lib/puppet/http/client.rb:31:in `connect'
     # ./spec/unit/http/client_spec.rb:75:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:180:in `block (2 levels) in <top (required)>'
     # util/rspec_runner:44:in `run'
     # util/rspec_runner:59:in `<main>'
```

Reply:
The `eq` was masking a problem, so when I moved to `equal` it failed. Turned out the client was completely ignoring the system ssl that was passed into the client. I updated that, and it should be good now.